account_banking_pain_base/models/account_payment_order.py

```diff
@@ -529,51 +529,89 @@ def generate_party_acc_number(
-                adrline1 = etree.SubElement(postal_address, "AdrLine")
-                adrline1.text = self._prepare_field(
-                    "Adress Line1",
-                    "partner.street",
+        if (
+            gen_args.get("pain_flavor").startswith("pain.001.001.")
+            or gen_args.get("pain_flavor").startswith("pain.008.001.")
+        ) and (partner.zip or partner.city):
+            adrline2 = etree.SubElement(postal_address, "AdrLine")
+            if partner.zip:
+                val = self._prepare_field(
+                    "zip",
+                    "partner.zip",
                     {"partner": partner},
                     70,
                     gen_args=gen_args,
                 )
-            if (
-                gen_args.get("pain_flavor").startswith("pain.001.001.")
-                or gen_args.get("pain_flavor").startswith("pain.008.001.")
-            ) and (partner.zip or partner.city):
-                adrline2 = etree.SubElement(postal_address, "AdrLine")
-                if partner.zip:
-                    val = self._prepare_field(
-                        "zip",
-                        "partner.zip",
-                        {"partner": partner},
-                        70,
-                        gen_args=gen_args,
-                    )
-                else:
-                    val = ""
-                if partner.city:
-                    val += " " + self._prepare_field(
-                        "city",
-                        "partner.city",
-                        {"partner": partner},
-                        70,
-                        gen_args=gen_args,
-                    )
-                adrline2.text = val
+            else:
+                val = ""
+            if partner.city:
+                val += " " + self._prepare_field(
+                    "city",
+                    "partner.city",
+                    {"partner": partner},
+                    70,
+                    gen_args=gen_args,
+                )
+            adrline2.text = val
         return True
 
+    @api.model
+    def _get_bank_record(self):
+        """Retrieve bank"""
+        bank = None
+        bank_id = self.env.context.get("export_bank_id")
+        if bank_id:
+            bank = self.env["res.bank"].browse(bank_id)
+        else:
+            po_id = self.env.context.get("payment_order_id")
+            if po_id:
+                po = self.env["account.payment.order"].browse(po_id)
+                bank = po.journal_id.bank_account_id.bank_id
+        return bank or self.env["res.bank"]
+
+    def generate_payment_file(self):
+        """Inject bank_id in context"""
+        self.ensure_one()
+        bank = (
+            self.company_partner_bank_id.bank_id
+            or self.journal_id.bank_account_id.bank_id
+        )
+        if bank:
+            self = self.with_context(export_bank_id=bank.id)
+        return super().generate_payment_file()
```

Comment:
@ntsirintanis This is not reached when using the [account_banking_sepa_credit_transfer](https://github.com/OCA/bank-payment/blob/16.0/account_banking_sepa_credit_transfer/models/account_payment_order.py#L17) or the [account_banking_sepa_direct_debit](https://github.com/OCA/bank-payment/blob/16.0/account_banking_sepa_direct_debit/models/account_payment_order.py#L17).

It would be better to inject the `bank_id` to `open2generated()` instead.

Reply:
Note that this is an api.model so self may or may not be an empty recordset